src/ucp/core/ucp_context.c

```diff
@@ -792,45 +796,35 @@ static unsigned ucp_tl_alias_count(ucp_tl_alias_t *alias)
 }
 
 static int ucp_tls_array_is_present(const char **tls, unsigned count,
-                                    const char *tl_name, const char *info,
-                                    uint8_t *rsc_flags, uint64_t *tl_cfg_mask)
+                                    const char *tl_name, int check_aux,
+                                    uint64_t *tl_cfg_mask)
 {
     uint64_t mask;
 
-    if ((mask = ucp_str_array_search(tls, count, tl_name, NULL)) != 0) {
-        *tl_cfg_mask |= mask;
-        ucs_trace("enabling tl '%s'%s", tl_name, info);
-        return 1;
-    } else if ((mask = ucp_str_array_search(tls, count, tl_name, "aux")) != 0) {
-        /* Search for tl names with 'aux' suffix, such tls can be
-         * used for auxiliary wireup purposes only */
-        *rsc_flags   |= UCP_TL_RSC_FLAG_AUX;
-        *tl_cfg_mask |= mask;
-        ucs_trace("enabling auxiliary tl '%s'%s", tl_name, info);
-        return 1;
-    } else {
-        return 0;
-    }
+    mask          = ucp_str_array_search(tls, count, tl_name,
+                                         check_aux ? UCP_TL_AUX_SUFFIX : NULL);
+    *tl_cfg_mask |= mask;
+    return (mask != 0);
 }
 
 static int
-ucp_config_is_tl_name_present(const ucs_config_allow_list_t *allow_list,
-                              const char *tl_name, int is_alias,
-                              uint8_t *rsc_flags, uint64_t *tl_cfg_mask)
+ucp_config_is_tl_name_present(const ucs_config_names_array_t *tl_array,
+                              const char *tl_name, int is_alias, int check_aux,
+                              uint64_t *tl_cfg_mask)
 {
     char strict_name[UCT_TL_NAME_MAX + 1];
 
     snprintf(strict_name, sizeof(strict_name), "\\%s", tl_name);
```

Comment:
@yosefe 
Should we keep this feature (\\\\)?
If I understand right, this feature was introduced to resolve the issue when there are the same alias names and transport names.

Reply:
I prefer to keep it. any reason to remove?

Reply:
Ok.
There are no serious reasons. Just thought about simplifying the code.